### PR TITLE
Add authentication with payload credentials

### PR DESCRIPTION
**Pull request by rshu-bd**
# Summary

This PR add authentication with payload credentials in addition to using environment variables and ROS params.  This PR is required by [PR#90](https://github.com/bdaiinstitute/spot_ros2/pull/90) in `spot_ros2` repo.

### CODE REVIEW

**Comment by heuristicus** [comment]
It looks like the authentication with payload credentials doesn't need the username or password, which makes those parameters unnecessary when calling the authenticate function with the credentials. This seems like it should be another function which is called when we receive payload credentials, to be more explicit about what is going on.

**Comment by rshu-bd** [comment]
Ready for a nother round of reviews.

**Review by jbarry-bdai — COMMENTED** [review]
Looks like maybe a lot of formatting changes crept into this?

File: spot_wrapper/wrapper.py [hunk 1]
```diff
@@ -5,58 +5,44 @@
...
-from bosdyn.client import frame_helpers
-from bosdyn.client import math_helpers
-from bosdyn.client import robot_command
+from bosdyn.api import (arm_command_pb2, geometry_pb2, image_pb2,
+                        manipulation_api_pb2, robot_command_pb2,
+                        robot_state_pb2, synchronized_command_pb2,
+                        trajectory_pb2)
+from bosdyn.api.graph_nav import graph_nav_pb2, map_pb2, nav_pb2
+from bosdyn.client import (ResponseError, RpcError, create_standard_sdk,
+                           frame_helpers, math_helpers, robot_command)
```
Comment (heuristicus) [inline spot_wrapper/wrapper.py:19]: This would be the correct way to do this, but let's keep that to another PR.

**Review by jbarry-bdai — COMMENTED** [review]
This looks pretty harmless but maybe we want to just make sure robots boot and authenticate in ROS1 (@heuristicus ) and ROS2 (@bhung-bdai @mpickett-bdai @amessing-bdai )?

**Comment by rshu-bd** [comment]
> Reference
Yes, working on it now.

**Comment by heuristicus** [comment]
> This looks pretty harmless but maybe we want to just make sure robots boot and authenticate in ROS1 (@heuristicus ) and ROS2 (@bhung-bdai @mpickett-bdai @amessing-bdai )?

I think I'm happy to merge as is since there aren't any changes to the basic authentication method. How are payload credential files generated?

**Comment by rshu-bd** [comment]
Is there a hook to run the formatting/linter locally?

**Comment by heuristicus** [comment]
> Is there a hook to run the formatting/linter locally?

I think you can use the black hook in  https://github.com/heuristicus/spot_ros2/commit/5b40d4997d000d82202457af1bc7d8aad5c948a8.

**Review by amessing-bdai — APPROVED** [review]

**Comment by amessing-bdai** [comment]
Tested
